spec/services/shared_examples_preserved_object_handler.rb

```diff
@@ -278,8 +335,10 @@
       ]
       obj_version_results = results.select { |r| codes.include?(r.keys.first) }
       msgs = obj_version_results.map { |r| r.values.first }
-      expect(msgs).to include(a_string_matching("PreservedObject"))
-      expect(msgs).to include(a_string_matching("PreservedCopy"))
+      unless results.find { |r| r.keys.first == PreservedObjectHandlerResults::INVALID_MOAB }
+        expect(msgs).to include(a_string_matching("PreservedObject"))
+        expect(msgs).to include(a_string_matching("PreservedCopy"))
+      end
```

Comment:
random observation:  i'm starting to feel just a bit uncomfortable with the amount of conditional logic and complexity creeping into the tests, but i don't have a good suggestion for how to go about things differently without making our already voluminous test code even more voluminous.  e.g., i would not go back to not having shared examples.

Reply:
completely agree.  Also had no handy solution short of a big refactoring which feels non-critical path.

Eventually we could probably morph most of the logging tests to something appropriate in the results_spec, and perhaps test for pohandler results directly, which might help a bit.

I agree the tests (and the code) are ugly as they are now.